server/src/service.rs

```diff
@@ -458,4 +474,42 @@ impl Service {
 
         Ok(signal_emitter)
     }
+
+    pub async fn remove_client(&self, old_owner: &str) -> Option<OwnedObjectPath> {
+        self.clients.lock().await.remove(old_owner)
+    }
+
+    pub async fn cleanup_sessions_on_client_disconnect(&self) -> Result<(), Error> {
+        let rule = zbus::MatchRule::builder()
+            .msg_type(zbus::message::Type::Signal)
+            .sender("org.freedesktop.DBus")?
+            .member("NameOwnerChanged")?
+            .build();
+        let mut stream = zbus::MessageStream::for_match_rule(rule, &self.connection, None).await?;
+
+        while let Some(message) = stream.try_next().await? {
+            let Ok((_name, old_owner, new_owner)) =
+                message.body().deserialize::<(String, String, String)>()
+            else {
+                continue;
+            };
+
```

Comment:
Is it possible that the owner changes? if not, maybe assert new_owner is NULL (once yo uupdate the matching rule)

Reply:
now, new_owner is NULL is handled by `.arg(2, "")?`